### DISCUSSION TITLE
Does Hydra provide phone/OTP login out of the box?

### DISCUSSION THREAD

**Discussion by Nefcanto** [ory/hydra#3758] [Q&A]
We're using Keycloak. Yet it does not have a default built-in phone/OTP login out of the box.
I read your tutorials and watched a couple of videos, but I could not find the answer to this question.
If our team migrates from Keycloak to Hydra, will we be able to have a simple phone/OTP login form?

**Comment by vinckr** [comment]
Hello @Nefcanto 

The service you are looking for is Ory Kratos: https://github.com/ory/kratos

- OTP: https://www.ory.sh/docs/kratos/passwordless/one-time-code
- SMS is currently available for MFA and verification. We are also working to add it as login method: https://www.ory.sh/docs/kratos/emails-sms/sending-sms
